File: Practicas/Qytetet/QytetetRuby/lib/casilla.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

require_relative "tipo_casilla"

module ModeloQytetet
  class Casilla
    attr_reader :numero_casilla, :coste, :titulo, :tipo
    def initialize(n, c)
      @numero_casilla = n
      @coste = c
    end
    def soy_edificable
      @tipo == TipoCasilla::CALLE 
    end
    
    def to_s
      "Casilla{numero_casilla=#{@numero_casilla}, coste=#{@coste}"
    end
  end
end

File: Practicas/Qytetet/QytetetJava/src/modeloqytetet/Casilla.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package modeloqytetet;

/**
 *
 * @author maxigang
 */
public class Casilla {
// Se podría haber hecho con una clase abstracta también    
    private int numeroCasilla, coste;
    
    public Casilla(int numeroCasilla, int coste)
    {
        this.numeroCasilla = numeroCasilla;
        this.coste = coste;
    }
    
    public int getNumeroCasilla() {
        return numeroCasilla;
    }

    int getCoste() {
        return coste;
    }

    protected TipoCasilla getTipo() {
        return null;
    }

    protected TituloPropiedad getTitulo() {
        return null;
    }
    
    protected void setCoste(int coste)
    {
        this.coste = coste;
    }
    
    protected boolean soyEdificable(){
        return false;
    }
    
    @Override
    public String toString() {
        String salida = "Casilla{" + "numeroCasilla=" + numeroCasilla + ", coste=" + 
                coste;
        return salida;
    }
    
}
